### PR TITLE
CDC ignore column/table/database name case

### PR DESCRIPTION
**Pull request by humengyu2012**
## What is the purpose of the change

Ignore column case in CDC.

## Brief change log

1. Change column name to lower case in craft and json.
2. Use CDC original column name in canal-json.

## Verifying this change

Use current tests.

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts:

- Dependencies (does it add or upgrade a dependency):  no
- The public API: no
- The runtime per-record code paths (performance sensitive): yes

## Documentation

- Does this pull request introduce a new feature? no
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by humengyu2012** [comment]
https://github.com/tidb-incubator/TiBigData/issues/250

File: flink/flink-1.13/src/main/java/io/tidb/bigdata/flink/format/cdc/TiDBCanalJsonDeserializationSchema.java [hunk 1]
```diff
@@ -44,6 +51,7 @@ public final class TiDBCanalJsonDeserializationSchema implements Deserialization
   private static final long serialVersionUID = 1L;
 
   private static final String FIELD_OLD = "old";
+  private static final String FIELD_DATA = "data";
```
Comment (xuanyu66) [inline flink/flink-1.13/src/main/java/io/tidb/bigdata/flink/format/cdc/TiDBCanalJsonDeserializationSchema.java:54]: unused field

**Review by xuanyu66 — CHANGES_REQUESTED** [review]
Plz add some tests

**Review by xuanyu66 — APPROVED** [review]

File: flink/flink-1.13/src/test/java/io/tidb/bigdata/flink/format/cdc/TiDBCanalJsonDeserializationSchemaTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    Assert.assertEquals(genericRowData, rowData);
+
+    // jsonDeserializer should be changed if columns are not match lowercase
+    Assert.assertEquals(ImmutableList.of("id", "Name"), fieldNames);
+    Assert.assertNotSame(jsonDeserializer, deserializationSchema.getJsonDeserializer());
+    jsonDeserializer = deserializationSchema.getJsonDeserializer();
+
+    collector.getRows().clear();
+
+    // jsonDeserializer should be only changed once
```
Comment (xuanyu66) [inline flink/flink-1.13/src/test/java/io/tidb/bigdata/flink/format/cdc/TiDBCanalJsonDeserializationSchemaTest.java:89]: What do you mean here？

**Review by xuanyu66 — CHANGES_REQUESTED** [review]
Plz add some comments

**Review by xuanyu66 — APPROVED** [review]

**Pull request merged**